backend/plugins/authorization.js

```diff
@@ -0,0 +1,68 @@
+var errors = require('restify-errors');
+
+import {getSetting} from '../settings.js';
+import {generateAndSaveAccessToken} from '../utils/authUtils';
+import fetch from 'node-fetch';
+
+const ESCAPED_ROUTES = [
+  new RegExp('^/$'),
+  new RegExp('^/login$'),
+  new RegExp('^/static/'),
+  new RegExp('^/oauth2')
+]
+
+function accessTokenIsValid(access_token) {
+    return getSetting('ACCESS_TOKEN') === access_token;
+}
+
+let win = '';
+
+export function PlotlyOAuth(electronWindow) {
+    win = electronWindow;
+
+    function isAuthorized(req, res, next) {
+        let path = req.href();
+
+        if (!getSetting('AUTH_ENABLED')) {
+          return (next());
+        }
+        // Auth is disabled for certain urls:
+        if ( ESCAPED_ROUTES.some(path.match.bind(path)) ) {
+          return (next());
+        }
+
+        // No Auth for electron apps:
+        if (win) {
+          return (next());
+        }
+        if (!accessTokenIsValid(req.cookies['db-connector-auth-token'])) {
+
+            if (!req.cookies['plotly-auth-token']) {
+              return (next(new errors.InvalidCredentialsError('Please login to access this page.')));
```

Comment:
this error handling is different than how we do error handling in the rest of the routes. I'd prefer that we'd use one error handling library. since we aren't passing this request forward to anything else, can this just be:
```
res.json(401, {error: {message: 'Please login to access this page'}});
return;
```